rules/aws_iam_policy_invalid_length.go

```diff
@@ -0,0 +1,61 @@
+package rules
+
+import (
+	"fmt"
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsIAMPolicyInvalidLengthRule checks that the policy length is less than 2048 characters
+type AwsIAMPolicyInvalidLengthRule struct {
+	resourceType  string
+	attributeName string
+}
+
+// NewAwsIAMPolicyInvalidLengthRule returns new rule with default attributes
+func NewAwsIAMPolicyInvalidLengthRule() *AwsIAMPolicyInvalidLengthRule {
+	return &AwsIAMPolicyInvalidLengthRule{
+		resourceType:  "aws_iam_policy",
+		attributeName: "policy",
+	}
+}
+
+// Name returns the rule name
+func (r *AwsIAMPolicyInvalidLengthRule) Name() string {
+	return "aws_iam_policy_invalid_length"
```

Comment:
Personally, I prefer names like `aws_iam_policy_too_long_policy` for such rules. Naming such as `{resource}_invalid_{attribute}` is for automatic generation, and it is better to have an obvious name for the violation.

Reply:
I will keep that in mind, and depending on where we go with this I will update it accordingly.